hack/e2e.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -o errexit   # exits immediately on any unexpected error (does not bypass traps)
+set -o nounset   # will error if variables are used without first being defined
+set -o pipefail  # any non-zero exit code in a piped command causes the pipeline to fail with that code
+
+KIND_VERSION="v0.5.1"
+
+start_docker() {
+   service docker start
+   # the service can be started but the docker socket not ready, wait for ready
+   WAIT_N=0
+   MAX_WAIT=5
+   while true; do
+      # docker ps -q should only work if the daemon is ready
+      docker ps -q > /dev/null 2>&1 && break
+      if [ ${WAIT_N} -lt ${MAX_WAIT} ]; then
+         WAIT_N=$((WAIT_N+1))
+         echo "Waiting for docker to be ready, sleeping for ${WAIT_N} seconds."
+         sleep ${WAIT_N}
+      else
+         echo "Reached maximum attempts, not waiting any longer..."
+         break
+      fi
+   done
+}
+
+install_govc() {
+   GOVC_VERSION=v0.21.0
+   GOVC_PKG_NAME=govc_linux_amd64
+   curl -L -O https://github.com/vmware/govmomi/releases/download/"${GOVC_VERSION}"/"${GOVC_PKG_NAME}".gz
+   gunzip "${GOVC_PKG_NAME}".gz
+   mv "${GOVC_PKG_NAME}" /usr/local/bin/govc
+   chmod +x /usr/local/bin/govc
+}
+
+on_exit() {
+   docker rm -f vpn
+}
+
+trap on_exit EXIT
+
+start_docker
+
+install_govc
+
+# install Kind
+wget "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-linux-amd64" \
+   --no-verbose -O /usr/local/bin/kind
+chmod +x /usr/local/bin/kind
+
+# install ginkgo
+GO111MODULE="on" go get github.com/onsi/ginkgo/ginkgo
+GO111MODULE="on" go get github.com/onsi/gomega/...
+
+export VSPHERE_SERVER="${GOVC_URL}"
+export VSPHERE_USERNAME="${GOVC_USERNAME}"
+export VSPHERE_PASSWORD="${GOVC_PASSWORD}"
+export VSPHERE_DATACENTER="SDDC-Datacenter"
+export VSPHERE_FOLDER="clusterapi"
+export VSPHERE_RESOURCE_POOL="clusterapi"
+export VSPHERE_DATASTORE="WorkloadDatastore"
+export VSPHERE_NETWORK="sddc-cgw-network-5"
+export VSPHERE_MACHINE_TEMPLATE="ubuntu-1804-kube-v1.16.2"
+export VSPHERE_HAPROXY_TEMPLATE="capv-haproxy-v0.5.3-77-g224e0ef6"
+
+export CAPI_IMAGE="gcr.io/k8s-staging-cluster-api/cluster-api-controller:v20200103-v0.2.5-497-gdbe789259"
+export CAPI_GIT_REF="09949bd397eecbfeac4e011b0d2c29fdbf2ac1ef"
+
+# Push new container images
+# TODO the `-k` flag here is a workaround until we can set GCR_KEY_FILE properly
+hack/release.sh -t pr -p
```

Comment:
Hi @yastij,

We don't need to push any images since the e2e test doesn't pivot. Everything is loaded locally into the Kind cluster.